client/lib/cgutil/cpuset_manager_linux_test.go

```diff
@@ -0,0 +1,164 @@
+package cgutil
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"testing"
+
+	"github.com/hashicorp/nomad/lib/cpuset"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+
+	"github.com/hashicorp/nomad/helper/uuid"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/nomad/helper/testlog"
+)
+
+func tmpCpusetManager(t *testing.T) (manager *cpusetManager, cleanup func()) {
+	if runtime.GOOS != "linux" || syscall.Geteuid() != 0 {
```

Comment:
We have this issue open https://github.com/hashicorp/nomad/issues/10051 about not doing a check for the UID but instead checking capabilities. This is "just test code" but it might be nice if we set the right example code here; if you were to run this code in an unprivileged Docker container as root I imagine it would still fail?

Reply:
Awesome, nice work on tracking all those down. I think cgroupfs just needs `CAP_SYSADMIN` correct? Do we have separate helpers for this yet?

Reply:
> I think cgroupfs just needs CAP_SYSADMIN correct?

I just double-checked and annoyingly it's not the manpage. But I believe that should do it.

> Do we have separate helpers for this yet?

Not yet, that's the first step in #10051.

Reply:
Given that there's no helper yet, maybe it'd be fine to punt on it so we can get this merged.